src/renderer/extensions/minimap/composables/useMinimapGraph.ts

```diff
@@ -96,35 +101,44 @@ export function useMinimapGraph(
     let positionChanged = false
     let connectionChanged = false
 
-    if (g._nodes.length !== lastNodeCount.value) {
+    // Use unified data source for change detection
+    const dataSource = MinimapDataSourceFactory.create(g)
+
+    // Check for node count changes
+    const currentNodeCount = dataSource.getNodeCount()
+    if (currentNodeCount !== lastNodeCount.value) {
       structureChanged = true
-      lastNodeCount.value = g._nodes.length
+      lastNodeCount.value = currentNodeCount
```

Comment:
Haven't looked into the factory yet, but this looks like it could use [usePrevious](https://vueuse.org/core/usePrevious/)